src/test/txreconciliation_tests.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/txreconciliation.h>
+
+#include <test/util/setup_common.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(txreconciliation_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(RegisterPeerTest)
+{
+    TxReconciliationTracker tracker(1);
+    const uint64_t salt = 0;
+
+    // Prepare a peer for reconciliation.
+    tracker.PreRegisterPeer(0);
+
+    // Both roles are false, don't register.
+    BOOST_CHECK(tracker.RegisterPeer(/*peer_id=*/0, /*is_peer_inbound=*/true,
```

Comment:
Here and elsewhere, `BOOST_CHECK_EQUAL(x, y)` is preferred over `BOOST_CHECK(x == y)` because the former gives a better message should the check fail.